vault/external_plugin_test.go

```diff
@@ -168,79 +143,75 @@ func TestCore_EnableExternalPlugin_MultipleVersions(t *testing.T) {
 		pluginType       consts.PluginType
 		registerVersions []string
 		mountVersion     string
+		expectedVersion  string
 		routerPath       string
 		expectedMatch    string
 	}{
 		"enable external credential plugin, multiple versions available": {
 			pluginType:       consts.PluginTypeCredential,
 			registerVersions: []string{"v1.0.0", "v1.0.1"},
 			mountVersion:     "v1.0.0",
+			expectedVersion:  "v1.0.0",
 			routerPath:       "auth/foo/bar",
 			expectedMatch:    "auth/foo/",
 		},
 		"enable external secrets plugin, multiple versions available": {
 			pluginType:       consts.PluginTypeSecrets,
 			registerVersions: []string{"v1.0.0", "v1.0.1"},
 			mountVersion:     "v1.0.0",
+			expectedVersion:  "v1.0.0",
 			routerPath:       "foo/bar",
 			expectedMatch:    "foo/",
 		},
 		"enable external credential plugin, multiple versions available, select other version": {
 			pluginType:       consts.PluginTypeCredential,
 			registerVersions: []string{"v1.0.0", "v1.0.1"},
 			mountVersion:     "v1.0.1",
+			expectedVersion:  "v1.0.1",
 			routerPath:       "auth/foo/bar",
 			expectedMatch:    "auth/foo/",
 		},
 		"enable external secrets plugin, multiple versions available, select other version": {
 			pluginType:       consts.PluginTypeSecrets,
 			registerVersions: []string{"v1.0.0", "v1.0.1"},
 			mountVersion:     "v1.0.1",
+			expectedVersion:  "v1.0.1",
+			routerPath:       "foo/bar",
+			expectedMatch:    "foo/",
+		},
```

Comment:
I suppose to round these out we could add similar tests for `mountVersion: "v1.0.0", expectedVersion: "v1.0.0"` to demonstrate picking the non-default version when there's multiple.

Reply:
I think that's covered in the first 2 cases? L153-154 and L161-162, LMK if I've misunderstood though.

Reply:
Oops, you're right, not sure what I was looking at 😃